test/signalrclienttests/websocket_transport_tests.cpp

```diff
@@ -478,10 +423,12 @@ void receive_loop_logs_exception_runner(const T& e, const std::string& expected_
     });
     mre.get();
 
-    receive_event.wait();
+    ASSERT_FALSE(client->receive_loop_started.wait(5000));
+
+    client->receive_message(std::make_exception_ptr(e));
 
     // this is race'y but there is nothing we can block on
-    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Can we check the following assertion in a loop until it succeeds or fails enough retries to make this less racy? Or can we use a test log writer that trips a cancellation_token once the expected log is logged?

Reply:
Sure, I'll attempt to make it a little better